packaging/os/zypper.py

```diff
@@ -135,8 +135,23 @@ def get_current_version(m, packages):
     return current_version
 
 
+def names_to_paths(names, paths):
+    """
+    Map package names to paths if these are URLs/local paths but
+    use the package name otherwise
+    """
+    packages = []
+    for p in names:
+        if paths.has_key(p):
```

Comment:
better use "in" then has_key() to be compatibel with python3
https://docs.python.org/3.1/whatsnew/3.0.html#builtins